server/chat-commands/moderation.ts

```diff
@@ -1930,25 +1936,94 @@ export const commands: ChatCommands = {
 		Punishments.addSharedIp(ip, note);
 		note = ` (${note})`;
 
-		this.addGlobalModAction(`The IP '${ip}' was marked as shared by ${user.name}.${note}`);
+		this.privateGlobalModAction(`The IP '${ip}' was marked as shared by ${user.name}.${note}`);
 		this.globalModlog('SHAREDIP', ip, ` by ${user.name}${note}`);
 	},
 	marksharedhelp: [
 		`/markshared [IP], [owner/organization of IP] - Marks an IP address as shared.`,
-		`Note: the owner/organization (i.e., University of Minnesota) of the shared IP is required. Requires @, &`,
+		`Note: the owner/organization (i.e., University of Minnesota) of the shared IP is required. Requires @ &`,
 	],
 
 	unmarkshared(target, room, user) {
 		if (!target) return this.parse('/help unmarkshared');
 		if (!this.can('globalban')) return false;
-		if (!/^[0-9.*]+$/.test(target)) return this.errorReply("Please enter a valid IP address.");
+		if (!IP_REGEX.test(target)) return this.errorReply("Please enter a valid IP address.");
 
 		if (!Punishments.sharedIps.has(target)) return this.errorReply("This IP isn't marked as shared.");
 
 		Punishments.removeSharedIp(target);
 
-		this.addGlobalModAction(`The IP '${target}' was unmarked as shared by ${user.name}.`);
+		this.privateGlobalModAction(`The IP '${target}' was unmarked as shared by ${user.name}.`);
 		this.globalModlog('UNSHAREIP', target, ` by ${user.name}`);
 	},
-	unmarksharedhelp: [`/unmarkshared [ip] - Unmarks a shared IP address. Requires @, &`],
+	unmarksharedhelp: [`/unmarkshared [IP] - Unmarks a shared IP address. Requires @ &`],
+
+	nomarkshared: {
+		add(target, room, user) {
+			if (!target) return this.parse(`/help nomarkshared`);
+			if (!(WHITELISTED_USERIDS.includes(user.id) || this.can('globalban'))) return false;
+			const [ip, ...reasonArr] = target.split(',');
+			if (!IP_REGEX.test(ip)) return this.errorReply(`Please enter a valid IP address.`);
+			if (!reasonArr?.length) {
+				this.errorReply(`Reasons are required.`);
+				this.parse(`/help nomarkshared`);
+				return;
+			}
+			if (Punishments.sharedIpBlacklist.has(ip)) {
+				return this.errorReply(`This IP is already blacklisted from being shared.`);
+			}
+			if (Punishments.sharedIps.has(ip)) this.parse(`/unmarkshared ${ip}`);
+			const reason = reasonArr.join(',');
+
+			Punishments.addBlacklistedSharedIp(ip, reason);
+
+			this.privateGlobalModAction(`The IP '${ip}' was blacklisted from being marked as shared by ${user.name}.`);
+			this.globalModlog('SHAREIP BLACKLIST', ip, ` by ${user.name}: ${reason.trim()}`);
+		},
+		remove(target, room, user) {
+			if (!target) return this.parse(`/help nomarkshared`);
+			if (!(WHITELISTED_USERIDS.includes(user.id) || this.can('globalban'))) return false;
+			if (!IP_REGEX.test(target)) return this.errorReply(`Please enter a valid IP address.`);
+			if (!Punishments.sharedIpBlacklist.has(target)) {
+				return this.errorReply(`This IP is not blacklisted from being shared.`);
```

Comment:
```suggestion
				return this.errorReply(`This IP is not blacklisted from being marked as shared.`);
```
(We can't stop people from sharing an IP, but we can stop PS staff from marking it as such.)